### DISCUSSION TITLE
Method Filament\Forms\Components\SpatieMediaLibraryFileUpload::getUploadedFileUsing does not exist.

### DISCUSSION THREAD

**Discussion by Alex92zz** [spatie/laravel-medialibrary#3342] [Q&A] [answered]
Hi, I have an error when trying to create a new post and I have this code "SpatieMediaLibraryFileUpload::make('avatar')," in the class PostResource extends Resource. 

my full code: "class PostResource extends Resource
{
    protected static ?string $model = Post::class;

    protected static ?string $navigationIcon = 'heroicon-o-collection';

    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Card::make()->schema([
                    Select::make('category_id')
                        ->relationship('Category', 'name')->required(),
                    TextInput::make('title')
                        ->reactive()
                        ->afterStateUpdated(function (Closure $set, $state) {
                            $set('slug', Str::slug($state));
                        })->required(),
                    TextInput::make('slug')->required(),
                    TextInput::make('description')->required(),
                    TextInput::make('keywords')->maxLength(255)->required(),
                    FileUpload::make('thumbnail'),
                    SpatieMediaLibraryFileUpload::make('avatar'),
                    RichEditor::make('content')->required(),
                    Toggle::make('is_published')
                ])
            ]);
    }" 

I did the import correctly: "use Filament\Forms\Components\SpatieMediaLibraryFileUpload;" 

If I delete it it works works well but I need it as I saw that it easy to upload images with it. 

my Post model: 
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Facades\Storage;
use Spatie\MediaLibrary\HasMedia;
use Spatie\MediaLibrary\InteractsWithMedia;

class Post extends Model implements HasMedia
{
    use HasFactory, InteractsWithMedia;

    protected $fillable = ['category_id', 'title', 'slug', 'content', 'description', 'keywords', 'is_published', 'thumbnail'];

    protected $casts = [
        'is_published' => 'boolean',
    ];

and I did the migration as it said on the website. 

Github project: 
https://github.com/Alex92zz/james-website-fillament

**Comment by dannyreinders** [comment, marked as answer]
Hi @Alex92zz 

I've had the same issue. I think you're using Filament V2 while the plugin uses functions that are only available in V3. Please try requiring the V2 version to solve your problem.

`composer require filament/spatie-laravel-media-library-plugin:"2.*" `

**Reply by Alex92zz** [reply]
Thank you for reply, it works now 👍
